### DISCUSSION TITLE
AI context provider! help!

### DISCUSSION THREAD

**Discussion by KolyaDychkevych** [vercel/next.js#67264] [Help]
### Summary

hi, this example works when i have app routing in my project
but when i have page routing, (i mean i have src folder in my project structure) not working.
where i should add AI wrapper ?

### Additional information

```js
import { ReactNode } from 'react';
import { AI } from './actions';

export default function RootLayout({
  children,
}: Readonly<{ children: ReactNode }>) {
  return (
    <html lang="en">
      <body>
        <AI>{children}</AI>
      </body>
    </html>
  );
}
```

**Comment by banuragaxioned** [comment]
This is what I did for the layout.tsx for the p the specific page.tsx where I was using RSC.
I didn't add it to the root layout, but I suppose that can work too (but might be unnecessary).

```js
import { AI } from "./actions";

export default async function DashboardLayout({ children }: { children?: React.ReactNode }) {
  return <AI>{children}</AI>;
}

```

**Reply by banuragaxioned** [reply]
More on this here: https://sdk.vercel.ai/docs/ai-sdk-rsc/generative-ui-state#using-ai--ui-state
